### PR TITLE
Adding a validate version error handling as discussed in issue #113

### PR DESCRIPTION
**Pull request by josecorella**
*Issue #, if available:*

*Description of changes:*

Issue #113
Added the version error handling.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-encryption-sdk-java.iml [hunk 1]
```diff
@@ -0,0 +1,215 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (lizroth) [inline aws-encryption-sdk-java.iml]: We probably don't want to commit this?

File: src/main/java/com/amazonaws/encryptionsdk/model/CiphertextHeaders.java [hunk 2]
```diff
@@ -179,6 +179,9 @@ public Boolean isComplete() {
      */
     private int parseVersion(final byte[] b, final int off) throws ParseException {
         version_ = PrimitivesParser.parseByte(b, off);
+        if(CiphertextType.deserialize(version_) == null) {
```
Comment (lizroth) [inline src/main/java/com/amazonaws/encryptionsdk/model/CiphertextHeaders.java]: Please add some tests

File: aws-encryption-sdk-java.iml (hunk 1 above)
Comment (mattsb42-aws) [reply aws-encryption-sdk-java.iml]: Not sure where this file came from, but if this is an IDE/build artifact, let's also add it to the `.gitignore`.

File: src/main/java/com/amazonaws/encryptionsdk/model/CiphertextHeaders.java (hunk 2 above)
Comment (mattsb42-aws) [inline src/main/java/com/amazonaws/encryptionsdk/model/CiphertextHeaders.java]: This is not the correct check.

We currently only have one valid version: `1`. Any other value MUST throw an error.

File: .gitignore [hunk 3]
```diff
@@ -5,3 +5,4 @@ target/
 .classpath
 /bin/
 .idea/
+*.iml/
```
Comment (mattsb42-aws) [inline .gitignore]: This is good, but `aws-encryption-sdk-java.iml` is still in the diff.

File: src/main/java/com/amazonaws/encryptionsdk/model/CiphertextHeaders.java [hunk 4]
```diff
@@ -179,6 +180,9 @@ public Boolean isComplete() {
      */
     private int parseVersion(final byte[] b, final int off) throws ParseException {
         version_ = PrimitivesParser.parseByte(b, off);
+        if(CiphertextType.deserialize(version_).getValue() != Byte.parseByte(VersionInfo.VERSION_NUM, 16)) {
```
Comment (mattsb42-aws) [inline src/main/java/com/amazonaws/encryptionsdk/model/CiphertextHeaders.java]: This is the major version of this implementation, not the ciphertext message format version. You want `CURRENT_CIPHERTEXT_VERSION`.
```suggestion
        if(CiphertextType.deserialize(version_).getValue() != Byte.parseByte(VersionInfo.CURRENT_CIPHERTEXT_VERSION, 16)) {
```

Comment (SalusaSecondus) [reply src/main/java/com/amazonaws/encryptionsdk/model/CiphertextHeaders.java]: Please actually change as follows: (note lack of parsing and space after `if`
```java
         if (CiphertextType.deserialize(version_).getValue() != VersionInfo.CURRENT_CIPHERTEXT_VERSION) { 
```

File: src/main/java/com/amazonaws/encryptionsdk/model/CiphertextHeaders.java [hunk 5]
```diff
@@ -179,6 +180,9 @@ public Boolean isComplete() {
      */
     private int parseVersion(final byte[] b, final int off) throws ParseException {
         version_ = PrimitivesParser.parseByte(b, off);
+        if(CiphertextType.deserialize(version_).getValue() != VersionInfo.CURRENT_CIPHERTEXT_VERSION ) {
```
Comment (SalusaSecondus) [inline src/main/java/com/amazonaws/encryptionsdk/model/CiphertextHeaders.java]: Our style requires a space between the `if` and before the `(`.

**Pull request merged**